neqo-transport/src/recovery.rs

```diff
@@ -320,32 +362,30 @@ impl LossRecovery {
         lost_packets
     }
 
-    fn set_loss_detection_timer(&mut self) {
-        qdebug!([self] "set_loss_detection_timer.");
-        let mut has_crypto_out = false;
-        let mut has_ack_eliciting_out = false;
+    fn get_loss_detection_timer(&self) -> (LossRecoveryMode, Option<Instant>) {
+        qdebug!([self] "get_loss_detection_timer.");
 
-        for pn_space in &[
+        let spaces = &[
             PNSpace::Initial,
             PNSpace::Handshake,
             PNSpace::ApplicationData,
-        ] {
-            let packet_space = &mut self.packet_spaces[*pn_space as usize];
+        ];
+        let mut has_crypto_out = false;
+        let mut has_ack_eliciting_out = false;
+        for sp in spaces
+            .iter()
+            .flat_map(|spc| self.spaces.get(*spc).sent_packets.values())
```

Comment:
Should the spaces have an iterator, so that you don't need to do this stuff?

Reply:
ok so we use this in two places, one of which (get_earliest_loss_time) needs to know which PNSpace the earliest value came from. So, going to try iterating PNSpace values instead of `LossRecoverySpace`s themselves.

If it were used more I could see also implementing the latter.